### ISSUE TITLE
Initializing js/ts features Causes Application to hang

### ISSUE THREAD

**Issue by jobu206** [microsoft/vscode#93268]
Issue Type: <b>Performance Issue</b>

Upon opening VS Code, JS/TS Features are initialized. This causes the Application to grind to a halt. I am not even using Typescript and I'm still forced to wait for this.

VS Code version: Code 1.43.1 (fe22a9645b44368865c0ba92e2fb881ff1afce94, 2020-03-18T07:17:38.324Z)
OS version: Linux x64 5.3.0-42-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz (4 x 3306)|
|GPU Status|2d_canvas: unavailable_software<br>flash_3d: unavailable_off<br>flash_stage3d: unavailable_off<br>flash_stage3d_baseline: unavailable_off<br>gpu_compositing: unavailable_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: unavailable_off<br>protected_video_decode: unavailable_off<br>rasterization: unavailable_off<br>skia_renderer: disabled_off_ok<br>video_decode: unavailable_off<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled_readback<br>webgl2: unavailable_off|
|Load (avg)|3, 1, 0|
|Memory (System)|15.58GB (8.92GB free)|
|Process Argv|--no-sandbox --unity-launch|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	   112	 18228	code main
    0	    32	 18232	   zygote
    0	   112	 18272	   gpu-process
    0	    48	 18284	   utility
    0	   112	 18496	   shared-process
    0	     0	 18893	     /bin/sh -c /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	     0	 18894	       /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    1	   255	 18612	   window (server.js - tasteslikehome - Visual Studio Code)
    0	   223	 18634	     extensionHost
    0	    80	 18707	       electron_node tsserver.js 
   35	   144	 18708	       electron_node tsserver.js 
    0	    80	 18776	         electron_node typingsInstaller.js typesMap.js 
    0	    96	 18744	       electron_node eslintServer.js 
    0	     0	 18642	     /bin/bash
   26	   303	 18847	       npm
    0	    48	 18689	     watcherService
    0	    48	 18706	     searchService
    0	    96	 18873	   window (Issue Reporter)
```

</details>
<details><summary>Extensions (17)</summary>

Extension|Author (truncated)|Version
---|---|---
jsrefactor|cms|2.20.5
bracket-pair-colorizer|Coe|1.0.61
vscode-eslint|dba|2.1.1
es7-react-js-snippets|dsz|2.7.0
vscode-html-css|ecm|0.2.3
vscode-great-icons|emm|2.1.47
prettier-vscode|esb|3.20.0
php-debug|fel|1.13.0
php-intellisense|fel|2.3.14
php-pack|fel|1.0.2
auto-rename-tag|for|0.1.1
python|ms-|2020.3.69010
vsliveshare|ms-|1.0.1783
LiveServer|rit|5.6.1
open-in-browser|tec|2.0.0
vscodeintellicode|Vis|1.2.6
vscode-icons|vsc|10.0.0

</details>

**Comment by mjbvz** [comment]
What exactly doesn't work while you see the progress indicator? It only indicates that the typescript server that powers both JavaScript and TypeScript language features is initializing (it is run its own process so it doesn't block VS Code)

**Comment by jobu206** [comment]
Well I can tell you that it is blocking VS Code from running. I can't do anything while that is running. Oftentimes it takes about 3-5 minutes to complete. Sometimes less, sometimes more.

**Comment by mjbvz** [comment]
But what does "can't do anything mean"? Does it not respond to any input?

**Comment by mjbvz** [comment]
Also does this reproduce with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled)?

**Comment by jobu206** [comment]
Ahh. Seems I may have spoken prematurely. Seems to not block actual functionality (browsing to various site files, etc). But won’t allow you to install any node packages until VS Code has completed the task in question. Is that expected? Haven’t had a chance to disable all extensions.

**Comment by mjbvz** [comment]
No, it should not block that. How are you installing node packages?

**Comment by jobu206** [comment]
I use NPM so... ``npm i <package name>``

**Comment by mjbvz** [comment]
Do you see an error during the install? Please also test with extensions disabled to make sure one of them isn't causing this

**Comment by jobu206** [comment]
Negative. No error seen. App just hangs while that task runs/completes.

**_My Repo Steps_**
- Open VS Code
- JS/TS Features are initialized
- Install a node package in a project via ``npm i abcPackage``
- App hangs until JS/TS Features task is complete

I'll try and test with extensions disabled and report back.

**Comment by mjbvz** [comment]
Did you see this after disabling extensions?

**Comment by mjbvz** [comment]
Can you please collect the TS Server log from when this happens:

1. Set `"typescript.tsserver.log": "verbose",`
1. Restart VS Code and reproduce the problem
1. In VS Code, run the `TypeScript: Open TS Server log` command
1. This should open a large log file called `tsserver.log` 

 If you can share the log, I can take a look to see if anything stands out that would explain this

 > **⚠️Warning:** The TypeScript log may include information from your workspace, including file paths and source code. If you have any concerns about posting this publicly on Github, just let me know and we can arrange something else. On our side, we only use these logs to investigate issues like this

**Comment by rynop** [comment]
I have the same problem. Vanilla es6 code (no typescript). Webpack & babel.

After "Initializing JS/TS language features" completes, intellisense works (f12/cmd+click, hover over imports etc).  The **BIG** problem however is, while "Initializing JS/TS language features" you can't even save a file (it gets put on the queue).

I REALLY want to use VS Code but this is a non-starter for me.

jsconfig.json

```json
{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "experimentalDecorators": true,
    "moduleResolution": "node",
    "jsx": "react",
    "checkJs": false,
    "baseUrl": "./",
    "paths": {
      // lots of entries here
      "SomeImport": [
        "/full/path/to/SomeImport.js"
      ],
    }
  },
  "exclude": [
    "build",
    "node_modules",
    "native_asset",
    "native_helpers",
    "node_modules_static_ext",
    "webpack_loaders",
    "webpack_plugins",
    "webpack_tools",
    "asset",
    "electron",
    "native_v2"
  ]
}
```

@mjbvz did you mean `"typescript.tsserver.trace": "verbose"`?  Also I don't see `TypeScript: Open TS Server log command` as a command option:
![image](https://user-images.githubusercontent.com/372730/79003249-d5d21280-7b17-11ea-9ec7-855a48b29d70.png)


If I use `"typescript.tsserver.trace": "verbose"` and open the `Output` console (and choose typescript in the dropdown) I get lots of output.

[Here is the log output](https://gist.github.com/rynop/4ef0e859d58626c27b1dfee9be3d68d4) - note the "Request took 63321 ms" - it is typically longer :(

The only extensions I have installed:
![image](https://user-images.githubusercontent.com/372730/79003349-05811a80-7b18-11ea-9465-8a2c7489c3fe.png)

**Comment by mjbvz** [comment]
@rynop Split your issue into #94943

**Comment by jobu206** [comment]
Hi Matt -
Here's the latest results.
[tsserver.log](https://github.com/microsoft/vscode/files/4471738/tsserver.log)
